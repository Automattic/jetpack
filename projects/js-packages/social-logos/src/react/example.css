.social-logos-example {
	max-width: 860px;
	color: #767676;
  	font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", "Roboto", "Oxygen-Sans", "Ubuntu", "Cantarell", "Helvetica Neue", sans-serif;
  	line-height: 1.15;
	margin: 20px auto;;
}

.display-control-group {
	display: flex;
	margin-bottom: 20px;
}

.display-control {
	display: flex;
	margin-right: 50px;
}

.display-control h4 {
	margin: 0 10px 0 0;
}

.icons {
	padding: 0 20px;
	overflow: hidden;
	margin-bottom: 50px;
	display: flex;
	flex-wrap: wrap;
	justify-content: center;
}

.icons div {
	width: 64px;
	height: 64px;
	float: left;
	padding: 6px 2px;
	position: relative;
	font-size: 7pt;
	cursor: pointer;
	text-align: center;
	margin-bottom: 20px;
}
