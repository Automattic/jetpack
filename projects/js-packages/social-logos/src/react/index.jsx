/**
 * Export components.
 */
export { default } from './social-logo';
export { SocialLogoData } from './social-logo-data';
