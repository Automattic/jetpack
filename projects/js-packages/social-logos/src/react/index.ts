/**
 * Export components.
 */
export * from './social-logo';
export { SocialLogo as default } from './social-logo';
export { SocialLogoData } from './social-logo-data';
