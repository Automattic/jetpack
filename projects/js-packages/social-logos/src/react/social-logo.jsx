import PropTypes from 'prop-types';
import React, { PureComponent } from 'react';
import { SocialLogoData } from './social-logo-data';

export default class SocialLogo extends PureComponent {
	static defaultProps = {
		size: 24,
	};

	static propTypes = {
		icon: PropTypes.string.isRequired,
		size: PropTypes.number,
		onClick: PropTypes.func,
		className: PropTypes.string,
	};

	render() {
		const { size, onClick, icon, className, ...otherProps } = this.props;

		const iconClass = [ 'social-logo', 'social-logo-' + icon, className ]
			.filter( Boolean )
			.join( ' ' );

		const logoData = SocialLogoData.find( logo => logo.name === icon );

		if ( ! logoData ) {
			return <svg height={ size } width={ size } { ...otherProps } />;
		}

		const svg = React.cloneElement( logoData.svg, {
			className: iconClass,
			height: size,
			width: size,
			onClick: onClick,
			...otherProps,
		} );
		return svg;
	}
}
