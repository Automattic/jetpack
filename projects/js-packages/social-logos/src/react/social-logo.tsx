import PropTypes from 'prop-types';
import React, { PureComponent } from 'react';
import { SocialLogoData } from './social-logo-data';

export type SocialLogoProps = React.SVGAttributes< SVGSVGElement > & {
	icon: ( typeof SocialLogoData )[ number ][ 'name' ];
	size?: number;
};

export class SocialLogo extends PureComponent< SocialLogoProps > {
	static defaultProps = {
		size: 24,
	};

	static propTypes = {
		icon: PropTypes.string.isRequired,
		size: PropTypes.number,
		onClick: PropTypes.func,
		className: PropTypes.string,
	};

	render() {
		const { size, onClick, icon, className, ...otherProps } = this.props;

		const iconClass = [ 'social-logo', 'social-logo-' + icon, className ]
			.filter( Boolean )
			.join( ' ' );

		const logoData = SocialLogoData.find( logo => logo.name === icon );

		if ( ! logoData ) {
			return <svg height={ size } width={ size } { ...otherProps } />;
		}

		const svg = React.cloneElement( logoData.svg, {
			className: iconClass,
			height: size,
			width: size,
			onClick: onClick,
			...otherProps,
		} );
		return svg;
	}
}
