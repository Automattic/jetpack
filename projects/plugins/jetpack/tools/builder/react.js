/**
 * External dependencies
 */
import log from 'fancy-log';
import gulp from 'gulp';
import PluginError from 'plugin-error';
import webpack from 'webpack';

function getWebpackConfig() {
	return [
		...require( './../webpack.config.js' ),
		...require( './../webpack.config.css.js' ),
		require( './../webpack.config.masterbar.js' ),
	];
}

export const watch = function () {
	const config = getWebpackConfig();

	return webpack( config ).watch(
		100,
		onBuild.bind( this, function ( error ) {
			if ( error ) {
				log( error );
				return;
			}
		} )
	);
};

gulp.task( 'react:master', function ( done ) {
	const config = getWebpackConfig();

	return webpack( config ).run( onBuild.bind( this, done ) );
} );

function onBuild( done, err, stats ) {
	// Webpack doesn't populate err in case the build fails
	// @see https://github.com/webpack/webpack/issues/708
	const erroringStats = stats.stats.find(
		( { compilation } ) => compilation.errors && compilation.errors.length
	);

	if ( erroringStats && done ) {
		done( new PluginError( 'webpack', erroringStats.compilation.errors[ 0 ] ) );
		return; // Otherwise gulp complains about done called twice
	}

	log(
		'Building JSâ€¦',
		stats.toString( {
			colors: true,
			hash: true,
			version: false,
			timings: true,
			assets: true,
			chunks: true,
			chunkModules: false,
			modules: false,
			cached: false,
			reasons: false,
			source: false,
			errorDetails: true,
			children: false,
		} ),
		'\nJS finished at',
		Date.now()
	);

	done();
}

export const build = gulp.series( 'react:master' );
