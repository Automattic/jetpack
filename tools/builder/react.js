/**
 * External dependencies
 */
import banner from 'gulp-banner';
import fs from 'fs';
import log from 'fancy-log';
import gulp from 'gulp';
import gulpif from 'gulp-if';
import PluginError from 'plugin-error';
import rename from 'gulp-rename';
import saveLicense from 'uglify-save-license';
import sourcemaps from 'gulp-sourcemaps';
import tap from 'gulp-tap';
import uglify from 'gulp-uglify';
import webpack from 'webpack';
import { JSDOM } from 'jsdom';

function getWebpackConfig() {
	return Object.create( require( './../../webpack.config.js' ) );
}

export const watch = function() {
	const config = getWebpackConfig();

	return webpack( config ).watch(
		100,
		onBuild.bind(
			this,
			function( error ) {
				if ( error ) {
					log( error );
					return;
				}
				buildStatic( function() {} );
			}
		)
	);
};

gulp.task( 'react:master', function( done ) {
	const config = getWebpackConfig();

	if ( 'production' !== process.env.NODE_ENV ) {
		config.plugins.push(
			new webpack.LoaderOptionsPlugin( {
				debug: true
			} )
		);
	}

	return webpack( config ).run(
		onBuild.bind(
			this,
			function( error ) {
				if ( error ) {
					done( error );
					return;
				}

				buildStatic( done );
			}
		)
	);
} );

function onBuild( done, err, stats ) {
	// Webpack doesn't populate err in case the build fails
	// @see https://github.com/webpack/webpack/issues/708
	if ( stats.compilation.errors && stats.compilation.errors.length ) {
		if ( done ) {
			done( new PluginError( 'webpack', stats.compilation.errors[ 0 ] ) );
			return; // Otherwise gulp complains about done called twice
		}
	}

	log( 'Building JS…', stats.toString( {
		colors: true,
		hash: true,
		version: false,
		timings: true,
		assets: true,
		chunks: true,
		chunkModules: false,
		modules: false,
		cached: false,
		reasons: false,
		source: false,
		errorDetails: true,
		children: false
	} ), '\nJS finished at', Date.now() );

	if ( 'production' === process.env.NODE_ENV ) {
		log( 'Uglifying JS...' );
		gulp.src( '_inc/build/admin.js' )
			.pipe( uglify() )
			.pipe( gulp.dest( '_inc/build' ) )
			.on( 'end', function() {
				log( 'Your JS is now uglified!' );
			} );
	}

	const is_prod = 'production' === process.env.NODE_ENV;

	const supportedModules = [
		'shortcodes',
		'widgets',
		'after-the-deadline',
		'widget-visibility',
		'custom-css',
		'publicize',
		'custom-post-types',
		'sharedaddy',
		'contact-form',
		'photon',
		'carousel',
		'related-posts',
		'tiled-gallery',
		'likes',
		'minileven',
		'infinite-scroll',
		'masterbar',
		'videopress',
		'comment-likes',
		'lazy-images'
	];

	// Source any JS for whitelisted modules, which will minimize us shipping much
	// more JS that we haven't pointed to in PHP yet.
	// Example output: modules/(shortcodes|widgets)/**/*.js
	const supportedModulesSource = `modules/@(${ supportedModules.join( '|' ) })/**/*.js`;

	// Uglify other JS from _inc and supported modules
	const sources = [
		'_inc/*.js',
		supportedModulesSource,
		'!modules/**/test-*.js',
	];

	// Don't process minified JS in _inc or modules directories
	const sourceNegations = [
		'!_inc/*.min.js',
		'!modules/**/*.min.js'
	];
	gulp.src( Array.concat( sources, sourceNegations ) )
		.pipe( banner( '/* Do not modify this file directly. It is compiled from other files. */\n' ) )
		.pipe( gulpif( ! is_prod, sourcemaps.init() ) )
		.pipe( uglify( {
			output: {
				comments: saveLicense
			}
		} ) )
		.pipe( rename( { suffix: '.min' } ) )
		.pipe( gulpif( ! is_prod, sourcemaps.write( 'maps' ) ) ) // Put the maps in _inc/build/maps so that we can easily .svnignore
		.pipe( gulp.dest( '_inc/build' ) )
		.on( 'end', function() {
			log( 'Your other JS is now uglified!' );
			done();
		} );
}

export const build = gulp.series( 'react:master' );

function buildStatic( done ) {
	log( 'Building static HTML from built JS…' );
	const { window } = new JSDOM();
	const { document } = ( new JSDOM( '' ) ).window;

	global.window = window;
	global.document = document;
	global.navigator = window.navigator;

	window.Initial_State = {
		dismissedNotices: [],
		connectionStatus: {
			devMode: {
				isActive: false
			}
		},
		userData: {
			currentUser: {
				permissions: {}
			}
		}
	};

	try {
		// normalize path
		const path = require.resolve( __dirname + '/../../_inc/build/static.js' );

		// Making sure NodeJS requires this file every time this is called
		delete require.cache[ path ];

		// Will throw when `path` does not exist, skipping file generation below that depends on `path`.
		require( path );

		gulp.src( [ '_inc/build/static*' ] )
			.pipe( tap( function( file ) {
				fs.unlinkSync( file.path );
			} ) )
			.on( 'end', function() {
				fs.writeFileSync( __dirname + '/../../_inc/build/static.html', window.staticHtml );
				fs.writeFileSync( __dirname + '/../../_inc/build/static-noscript-notice.html', window.noscriptNotice );
				fs.writeFileSync( __dirname + '/../../_inc/build/static-version-notice.html', window.versionNotice );
				fs.writeFileSync( __dirname + '/../../_inc/build/static-ie-notice.html', window.ieNotice );

				done();
			} );
	} catch ( error ) {
		done( error );
	}
}
