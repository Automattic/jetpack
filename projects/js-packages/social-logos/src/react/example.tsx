/* eslint-disable no-alert -- ok for demo */
import React, { useState } from 'react';
import { SocialLogo } from './social-logo';
import { SocialLogoData } from './social-logo-data';
import '../css/example.css';

/**
 * An example React component that displays all the social logos.
 *
 * @return {React.Component} The `SocialLogosExample` component.
 */
function SocialLogosExample() {
	const [ useSmallIcons, setUseSmallIcons ] = useState( false );
	const [ showIconNames, setShowIconNames ] = useState( true );

	const iconSize = useSmallIcons ? 24 : 48;

	const handleClick = name => {
		const code = `<SocialLogo icon="${ name }" size="${ iconSize }" />`;
		window.prompt( 'Copy component code:', code );
	};

	const handleSmallIconsToggle = e => {
		setUseSmallIcons( e.target.checked );
	};

	const handleIconNamesToggle = e => {
		setShowIconNames( e.target.checked );
	};

	const allSocialLogos = SocialLogoData.map( logo => {
		return (
			<div key={ logo.name }>
				<SocialLogo
					icon={ logo.name }
					size={ iconSize }
					onClick={ handleClick.bind( this, logo.name ) }
				/>
				{ showIconNames && <p>{ logo.name }</p> }
			</div>
		);
	} );

	return (
		<div className="social-logos-example">
			<h1>Social Logos</h1>

			<div className="display-control-group">
				<div className="display-control">
					<h4>Small icons</h4>
					{ /* eslint-disable-next-line jsx-a11y/label-has-associated-control -- https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/issues/869 */ }
					<label className="switch">
						<input type="checkbox" onChange={ handleSmallIconsToggle } checked={ useSmallIcons } />
						<span className="handle"></span>
					</label>
				</div>
				<div className="display-control">
					<h4>Icon names</h4>
					{ /* eslint-disable-next-line jsx-a11y/label-has-associated-control -- https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/issues/869 */ }
					<label className="switch">
						<input type="checkbox" onChange={ handleIconNamesToggle } checked={ showIconNames } />
						<span className="handle"></span>
						<span className="switch-label" data-on="On" data-off="Off"></span>
					</label>
				</div>
			</div>

			<div className="icons">{ allSocialLogos }</div>

			<p>
				<a href="https://github.com/Automattic/social-logos">GitHub</a>
			</p>
		</div>
	);
}

export default SocialLogosExample;
