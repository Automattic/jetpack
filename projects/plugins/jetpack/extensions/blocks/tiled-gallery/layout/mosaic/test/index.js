import { render } from '@testing-library/react';
import { range } from 'lodash';
import React from 'react';
import Mosaic from '..';
import * as imageSets from '../../test/fixtures/image-sets';

test.each( Object.entries( imageSets ) )( 'renders as expected (set %s)', ( k, images ) => {
	const { container } = render(
		<Mosaic images={ images } renderedImages={ range( images.length ) } />
	);
	expect( container ).toMatchSnapshot( 'images' );
} );
