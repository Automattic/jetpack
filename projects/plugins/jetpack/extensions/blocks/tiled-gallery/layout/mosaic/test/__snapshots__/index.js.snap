// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as expected (set imageSet1): images 1`] = `
<div>
  <div
    class="tiled-gallery__gallery"
  >
    <div
      class="tiled-gallery__row"
    >
      <div
        class="tiled-gallery__col"
      >
        0
      </div>
    </div>
    <div
      class="tiled-gallery__row"
    >
      <div
        class="tiled-gallery__col"
      >
        1
      </div>
    </div>
    <div
      class="tiled-gallery__row"
    >
      <div
        class="tiled-gallery__col"
      >
        2
      </div>
      <div
        class="tiled-gallery__col"
      >
        3
      </div>
      <div
        class="tiled-gallery__col"
      >
        4
      </div>
      <div
        class="tiled-gallery__col"
      >
        5
      </div>
    </div>
    <div
      class="tiled-gallery__row"
    >
      <div
        class="tiled-gallery__col"
      >
        6
      </div>
      <div
        class="tiled-gallery__col"
      >
        7
      </div>
    </div>
    <div
      class="tiled-gallery__row"
    >
      <div
        class="tiled-gallery__col"
      >
        8
      </div>
      <div
        class="tiled-gallery__col"
      >
        9
        10
      </div>
    </div>
    <div
      class="tiled-gallery__row"
    >
      <div
        class="tiled-gallery__col"
      >
        11
        12
      </div>
      <div
        class="tiled-gallery__col"
      >
        13
      </div>
    </div>
  </div>
</div>
`;
